source/camel-case.d.ts

```diff
@@ -0,0 +1,49 @@
+import {WordSeparators} from './delimiter-case';
+
+export type Split<S extends string, D extends string> =
+	string extends S ? string[] :
+	S extends '' ? [] :
+	S extends `${infer T}${D}${infer U}` ? [T, ...Split<U, D>] :
+	[S];
+
+type InnerCamelCaseStringArray<Parts extends any[], PreviousPart> =
+	Parts extends [`${infer FirstPart}`, ...infer RemainingParts]
+		? FirstPart extends undefined
+			? ''
+			: FirstPart extends ''
+					? InnerCamelCaseStringArray<RemainingParts, PreviousPart>
+					: `${PreviousPart extends '' ? FirstPart : Capitalize<FirstPart>}${InnerCamelCaseStringArray<RemainingParts, FirstPart>}`
+		: '';
+
+type CamelCaseStringArray<Parts extends string[]> =
+	Parts extends [`${infer FirstPart}`, ...infer RemainingParts]
+		? `${FirstPart}${InnerCamelCaseStringArray<RemainingParts, FirstPart>}`
+		: never;
+
+/**
+Converts a string literal from any typical non-camel-case casing to camel-case.
+
+This can be useful when, for example, converting some kebab-cased command-line flags or a snake-cased database result.
+
+@example
+```
+import {CamelCase} from 'type-fest';
+
+type CamelCasedProps<T> = {
```

Comment:
We can do this later, but I think this is an extremely common situation, so I think we should expose this as a type too.

Reply:
Very much agree that this would be a good next step